sparkly/session.py

```diff
@@ -187,11 +188,20 @@ def _setup_jars(self):
             return ''
 
     def _setup_options(self, additional_options):
-        options = list(self.options.items())
+        options = {}
+
+        options.update(self.options)
+
         if additional_options:
-            options += list(additional_options.items())
+            options.update(additional_options)
 
-        return sorted(options)
+        if 'spark.sql.catalogImplementation' not in options:
+            options['spark.sql.catalogImplementation'] = 'hive'
+
+        if options:
+            return ' '.join('--conf "{}={}"'.format(*o) for o in sorted(options.items()))
```

Comment:
@drudim had an offline discussion with @mantzouratos in which he explained that this is needed so the options provided through the environment variable override the ones hard-coded in the SparkSession. @mantzouratos : can you add to the code a comment on this?